README.md

```diff
@@ -33,6 +33,19 @@ Application Options:
   --sign-args                 Extra arguments to use when signing with gpg[2]
 ```
 
+
+### Environment variables
+
+Some of the parameters above can alternatively be specified as environment variables. Also, some additional environment variables are available, too.
+
+- `ARCH`: Needs to be set whenever appimagetool cannot automatically determine the architecture of the binaries inside the AppDir to choose a suitable runtime (e.g., when binaries for multiple architectures or just shell scripts are contained in there).
+- `APPIMAGETOOL_FORCE_SIGN`: By default, if signing fails, appimagetool just logs a warning but will not abort immediately. If this environment variable is set, appimagetool exits with a non-zero return code.
+- `APPIMAGETOOL_SIGN_PASSPHRASE`: If the `--sign-key` is encrypted and requires a passphrase to be used for signing (and, for some reason, GnuPG cannot be used interactively, e.g., in a CI environment), this environment variable can be used to safely pass the key.
+- `GITHUB_TOKEN`: If set to any value, appimagetool will try to guess some update information to the current repository's release page. Note that this guessing is opinionated and requires the releases to be set up in a specific way. If it is *not* set up, AppImageUpdate will try to update but always fail.
+- `CI_COMMIT_REF_NAME`: If this (and a few other) environment variables are set by GitLab CI, update information will be guessed in a way similar to the GitHub actions code. Again, this approach is opinionated and assumes a matching setup.
```

Comment:
If set to any value, appimagetool will try to guess the update information based on environment variables present in the GitLab pipeline runner environment. Note that in order for AppImageUpdate to work, the generated `.AppImage` and `.AppImage.zsync` files need to be uploaded as GitLab artifacts without being renamed.

Reply:
You do not have to repeat this more than once. We should move this to a separate section that explains this guess work.

Also, we really need to tell people that this only works when `-g` is passed.

Reply:
Good point. I think `-g` is what actually needs to be documented, not these variables; especially because the ones you have listed here are just part of the "guessing magic" but are not suffcient alone.

Reply:
Let's just not mention them at all then and improve `-g`'s docs. They should be considered an implementation detail.